### PR TITLE
Add a cache for FxA tokens validation.

### PR DESCRIPTION
**Pull request by Natim**
Refs #121

### CODE REVIEW

File: readinglist/authentication.py [hunk 1]
```diff
@@ -64,20 +64,25 @@ def _get_credentials(self, request):
...
+
+            auth_client = OAuthClient(server_url=server_url)
+            try:
+                profile = auth_client.verify_token(token=auth, scope=scope)
+                hmac_secret = settings.get('readinglist.userid_hmac_secret')
+                user_id = hmac.new(hmac_secret.encode('utf-8'),
+                                   profile['user'].encode('utf-8'),
+                                   hashlib.sha256).hexdigest()
+                request.registry.session.set('fxa-oauth-token.%s' % auth, user_id)
+                request.registry.session.expire('fxa-oauth-token.%s' % auth, 5 * 60)
```
Comment (Natim) [inline readinglist/authentication.py]: Needs to be in a settings.

**Comment by n1k0** [comment]
Tested with latest client, approved.

File: config/readinglist.ini [hunk 2]
```diff
@@ -15,6 +15,7 @@ fxa-oauth.client_id = 89513028159972bc
 fxa-oauth.client_secret = 9aced230585cc0aa2932e2eb871c9a3a7d6458e59ccf57eb610ea0a3467dd800
 fxa-oauth.oauth_uri = https://oauth-stable.dev.lcip.org
 fxa-oauth.scope = profile
+fxa-oauth.cache_verification = 300
```
Comment (ametaireau) [inline config/readinglist.ini]: cache_ttl

File: readinglist/authentication.py [hunk 3]
```diff
@@ -64,20 +64,33 @@ def _get_credentials(self, request):
...
-            profile = auth_client.verify_token(token=auth, scope=scope)
-            hmac_secret = settings.get('readinglist.userid_hmac_secret')
-            user_id = hmac.new(hmac_secret.encode('utf-8'),
-                               profile['user'].encode('utf-8'),
-                               hashlib.sha256).hexdigest()
-        except fxa_errors.OutOfProtocolError:
-            raise HTTPServiceUnavailable()
-        except (fxa_errors.InProtocolError, fxa_errors.TrustError):
-            return None
+        session = getattr(request.registry, 'session', None)
```
Comment (ametaireau) [inline readinglist/authentication.py]: can we pass the session directly when we're creating the authentication object?

File: config/readinglist.ini (hunk 2 above)
Comment (n1k0) [inline config/readinglist.ini]: Nit: it's often considered good practice to name the setting using the type of unit used; here, `cache_ttl_seconds` would be meaningful.

**Comment by leplatrem** [comment]
Small test to assert that fxa server is not hit during ttl ?

**Pull request closed without merging**